src/ua_connection.c

```diff
@@ -102,6 +108,8 @@ separateIncompleteChunk(UA_Connection *connection, UA_ByteString * UA_RESTRICT m
                 return retval;
             memcpy(connection->incompleteMessage.data, message->data, message->length);
             connection->releaseRecvBuffer(connection, message);
+            //The complete message is buffered, so no more content to be processed in message
+            message->length = 0;
```

Comment:
Yes, you are right, it was just not implemented for the testing network layer `releaseRecvBuffer`. I'll remove this change